sbtc/src/deposits.rs

```diff
@@ -128,14 +128,15 @@ impl CreateDepositRequest {
     /// This function fetches the transaction using the given client and
     /// checks that the transaction has been confirmed. The transaction
     /// need not be confirmed.
-    pub fn validate<C>(&self, client: &C) -> Result<Deposit, Error>
+    pub async fn validate<C>(&self, client: &C) -> Result<Deposit, Error>
     where
         C: BitcoinClient,
     {
         // Fetch the transaction from either a block or from the mempool
         let response = client
             .get_tx(&self.outpoint.txid)
-            .map_err(|err| Error::BitcoinClient(Box::new(err)))?;
+            .await
+            .map_err(|_| Error::GetBitcoinTx(self.outpoint.txid))?;
```

Comment:
Although I'd like to keep the actual error, I'm thinking that this function should be moved anyway since it's not really needed in this crate. But if this stays, we should try to keep the `Error::BitcoinClient` variant, and perhaps enhance it with the `txid` field.

Reply:
Was able to keep it :+1: